### ISSUE TITLE
Navigation links in `CollectionView`

### ISSUE THREAD

**Issue by danielsaidi** [defagos/SwiftUICollection#4]
Hi! I noticed that navigation links don't work when I use them within a `CollectionView`, while regular buttons do work. I wonder if it has something to do with the hosting controller and how it may block the navigation from happening.

**Comment by defagos** [comment]
Good catch! I wouldn't be surprised if this is related to missing parent-child containment relationship, see [this tweet from Peter Steinberger](https://twitter.com/steipete/status/1308142116132982784) and related articles and links.

Didn't have to focus on this specific question myself, but this is definitely where I would have a look first.

**Comment by danielsaidi** [comment]
I'd be happy to not have navigation links, but rather use buttons that trigger sheets or full screen covers...but I guess it would be nice to solve this for other scenarios than tvOS.

**Comment by danielsaidi** [comment]
Since the scroll view offset and item re-selection works when returning to the collection from a sheet or full screen modal (after I fixed my incorrect data reload), I have no need to do "real" navigation anymore...at least for now. It may however be good to have support for navigation links when the target destination can have deeper navigation, right?

**Comment by defagos** [comment]
I fear this collection view implementation might have some limitations, but if navigation links could be properly supported that would be nice (and probably something has to be learnt when trying to make them work).

I therefore labeled this issue as a bug.

**Comment by danielsaidi** [comment]
Perfect, I'll see if I can come up with a solution, but until then the sheet and full screen cover approach work well when the navigation is not that deep. I don't even know if there would be any problems with deep navigation and sheets on tvOS.

**Comment by defagos** [comment]
Didn’t see any deep navigation issues in the app I am working on 🙂

**Comment by danielsaidi** [comment]
Great! I realized that perhaps on tvOS the sheet implementation _is_ the same thing as pushing onto a navigation stack. I mean, there is no sheet concept on the platform, so the implementation could actually be that they do that.
